R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,456 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
+      left_join(L1322.Fert_NEcost_75USDkgN_F, by = c('technology' = 'fuel') ) %>%
+      rename(input.cost = NEcost_75USDkgN) %>%
+      mutate(input.cost = round(input.cost, energy.DIGITS_COST)) %>%
+      na.omit -> #Export technologies have no cost assigned. Just drop the object
+      L2322.GlobalTechCost_Fert
+
+    # Carbon capture rates from technologies with CCS
+    # L2322.GlobalTechCapture_Fert: CO2 capture fractions from global fertilizer production technologies with CCS
+    ## No need to consider historical periods or intermittent technologies here
+    A322.globaltech_co2capture %>%
+      gather(year, remove.fraction, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.numeric(year)) ->
+      A322.globaltech_co2capture_long
+
+    df_years <- unique(A322.globaltech_co2capture_long$year)
+    int_years <- setdiff(FUTURE_YEARS, df_years)
 
+    A322.globaltech_co2capture_long %>%
+      select(-remove.fraction, -year) %>%
+      unique() %>%
+      repeat_add_columns(tibble(year = int_years)) %>%
+      bind_rows(A322.globaltech_co2capture_long) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(remove.fraction = approx_fun(year, remove.fraction, rule = 1)) %>%
+      mutate(remove.fraction = round(remove.fraction, energy.DIGITS_REMOVE.FRACTION)) %>%
+      ungroup %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) %>%
+      mutate(storage.market = "carbon-storage") ->
+      L2322.GlobalTechCapture_Fert
+
+    # Retirement information
+    A322.globaltech_retirement %>%
+      set_years %>%
+      mutate(year = as.numeric(year)) %>%
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      A322.globaltech_retirement_with_years
+
+    # Copy the data in the first future period through to the end year
+    A322.globaltech_retirement_with_years %>%
+      filter(year == max(BASE_YEARS)) ->
+      A322.globaltech_retirement_max_baseyear
+
+    A322.globaltech_retirement_with_years %>%
+      filter(year == min(FUTURE_YEARS)) %>%
+      select(-year) %>%
+      repeat_add_columns(tibble(year = FUTURE_YEARS)) %>%
+      bind_rows(A322.globaltech_retirement_max_baseyear) ->
+      L2322.globaltech_retirement
+
+    # Retirement may consist of any of three types of retirement function (phased, s-curve, or none)
+    # All of these options have different headers, and all are allowed
+
+    # L2322.GlobalTechShutdown_Fert: Global tech lifetime and shutdown rate
+    tibble(x = NA) %>%
+      add_units("None") %>%
+      add_comments("Not generated") %>%
+      add_flags(FLAG_NO_TEST) ->
+      L2322.GlobalTechShutdown_Fert
+
+    # L2322.GlobalTechSCurve_Fert: Global tech lifetime and s-curve retirement function
+    L2322.globaltech_retirement %>%
+      filter(!is.na(half.life)) %>%
+      select(LEVEL2_DATA_NAMES[["GlobalTechYr"]], "lifetime", "steepness", "half.life") ->
```

Comment:
Probably need to change this `select(one_of(c(...)))` for dplyr 0.5 compatibility